File: spring-web-java/src/main/java/spring/web/java/domain/member/repository/MemberRepository.java
package spring.web.java.domain.member.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import spring.web.java.domain.member.Member;

public interface MemberRepository extends JpaRepository<Member, Long> {

	@Query("""
		SELECT m
		FROM Member m
		WHERE m.account = :account
		""")
	Optional<Member> findByAccount(String account);
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/member/repository/MemberRepository.kt
package spring.web.kotlin.domain.member.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import spring.web.kotlin.domain.member.Member

interface MemberRepository : JpaRepository<Member, Long> {

    @Query(
        """
            SELECT m
            FROM Member m
            WHERE m.account = :account
            """
    )
    fun findByAccount(account: String): Member?
}